src/OpenFeature.SDK/Model/EvaluationContext.cs

```diff
@@ -10,96 +8,175 @@ namespace OpenFeature.SDK.Model
+        /// Gets the Value at the specified key
         /// </summary>
-        /// <param name="key">Key</param>
-        /// <param name="value">Value</param>
-        /// <typeparam name="T">Type of value</typeparam>
-        public void Add<T>(string key, T value)
+        /// <param name="key"></param>
+        /// <returns></returns>
+        public Value GetValue(string key) => this._structure.GetValue(key);
+
+        /// <summary>
+        /// Bool indicating if the specified key exists in the evaluation context
+        /// </summary>
+        /// <param name="key"></param>
+        /// <returns></returns>
+        public bool ContainsKey(string key) => this._structure.ContainsKey(key);
+
+        /// <summary>
+        /// Removes the Value at the specified key
+        /// </summary>
+        /// <param name="key"></param>
+        public void Remove(string key) => this._structure.Remove(key);
+
+        /// <summary>
+        /// Gets the value associated with the specified key
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public bool TryGetValue(string key, out Value value) => this._structure.TryGetValue(key, out value);
+
+        /// <summary>
+        /// Gets all values as a Dictionary
+        /// </summary>
+        /// <returns></returns>
+        public IDictionary<string, Value> AsDictionary()
+        {
+            return new Dictionary<string, Value>(this._structure.AsDictionary());
+        }
+
+        /// <summary>
+        /// Add a new bool Value to the evaluation context
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public EvaluationContext Add(string key, bool value)
+        {
+            this._structure.Add(key, value);
+            return this;
+        }
+
+        /// <summary>
+        /// Add a new string Value to the evaluation context
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public EvaluationContext Add(string key, string value)
+        {
+            this._structure.Add(key, value);
+            return this;
+        }
+
+        /// <summary>
+        /// Add a new int Value to the evaluation context
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public EvaluationContext Add(string key, int value)
+        {
+            this._structure.Add(key, value);
+            return this;
+        }
+
+        /// <summary>
+        /// Add a new double Value to the evaluation context
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public EvaluationContext Add(string key, double value)

```

Comment:
In [java](https://github.com/open-feature/java-sdk/blob/d5bd0fbf02b00da71149bf99cf108e0020f0afa2/src/main/java/dev/openfeature/javasdk/EvaluationContext.java#L16), we use [lombok's delegate](https://projectlombok.org/features/experimental/Delegate) to automatically proxy these methods to the encapsulated `structure`. I would like to do something like that here, but I couldn't find any such solutions.

@kinyoklion maybe you have an idea?

Reply:
In regards to directly replicating what can be done with lombok, I do not think that there is a wide-spread or idiomatic equivalent. 

There are ways to accomplish it using something like fody/cecil, but I don't think that would be good for a library like this.

Reply:
Cool, I don't think it's a hard requirement, just wondering if you had a simple solution up your sleeve.